### PR TITLE
 new logic for tests - bad and good example; hook checking nb outputs;

### PR DESCRIPTION
**Pull request by AgnieszkaZaba**

### CODE REVIEW

**Review by Sfonxu — APPROVED** [review]

**Comment by slayoo** [comment]
To make the hooks maintainable, it would be great to store in the repo some example files that demonstrate when the hooks should fail.  Chat suggests the following pattern:
image
Could it work?

**Comment by Sfonxu** [comment]
> To make the hooks maintainable, it would be great to store in the repo some example files that demonstrate when the hooks should fail. Chat suggests the following pattern: [...] Could it work?

I think it could, with maybe a more clear naming convention (like tests/passing/(hook_name).py, tests/failing/(hook_name).py )? 

Same goes for example notebooks I suppose. And maybe a good idea would be to add a CI job to run the hooks on those files as well.

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -38,3 +38,12 @@ repos:
         args: [--fix-header, --repo-name=devops_tests]
         language: python
         types: [jupyter]
+        exclude: tests/examples/bad/
```
Comment (AgnieszkaZaba) [inline .pre-commit-config.yaml]: Needed to actually make commit. We can think about running this pre-commits only on CI.

**Comment by AgnieszkaZaba** [comment]
- [x] run pre-commit hooks on both paths (bad and good) with fail mark
- [x] add test with edge cases (injected cells in test)

File: .pre-commit-config.yaml (hunk 1 above)
Comment (slayoo) [reply .pre-commit-config.yaml]: like in the above suggestion?
image

Comment (AgnieszkaZaba) [reply .pre-commit-config.yaml]: Yes, but then we have to remove this hook from config because the suggested code will not run on excluded files.

**Review by slayoo — APPROVED** [review]
